### ISSUE TITLE
[REQUEST] Add a way to keep packages out of Conda Env file

### ISSUE THREAD

**Issue by bethune-bryant** [open-ce/open-ce#212]
**Is your feature request related to a problem? Please describe.**
Packages that are only needed at build time are being included in the generated conda env files.

**Describe the solution you'd like**
We can add a property to the feedstock object in the open-ce env files called `runtime_package` that defaults to True. This value being false will keep it out of the generated conda env file.

Non runtime packages should also be removed when performing validation to ensure that there isn't something needed that won't be included.

**Additional context**
Example:
```
packages:
  - feedstock : cdts
    runtime_package : False
  - feedstock : bazel
    runtime_package : False
```

Related to #194

**Comment by npanpaliya** [comment]
This is similar to what I'd observed earlier. See 1b from https://github.com/open-ce/open-ce/issues/45#issuecomment-702127729.

**Comment by npanpaliya** [comment]
@bethune-bryant - are you taking this up? Or I should combine this with #194 . 
For #194 - I've been able to reduce the duplicates which brings down dependencies' count from 142 to 87 for dali env (as an example). I'm removing duplicates like `tensorflow-base` when `tensorflow-base >=2.4.0` is already included Or `python` when `python >=3.6` is present.
